canal/sync.go

```diff
@@ -106,6 +106,23 @@ func (c *Canal) runSyncBinlog() error {
 				}
 			}
 			continue
+		case *replication.TransactionPayloadEvent:
+			// handle subevent row by row
+			ev := ev.Event.(*replication.TransactionPayloadEvent)
+			for _, subEvent := range ev.Events {
+				err = c.handleRowsEvent(subEvent)
```

Comment:
Thanks for contributing!
these lines seems almost same with line 97~105, do you have time to unify them? like create a closure for one `*replication.RowsEvent`

Reply:
I move some error check into handleRowsEvent, it may be less duplicated code